.github/workflows/upterm_debug/action.yml

```diff
@@ -0,0 +1,17 @@
+name: "Upterm for debugging tests"
```

Comment:
I would suggest moving this code into `.github/workflows/tmate_debug/action.yml` so that we have one file that defines our "interactive debugging". 

I think currently, having both upterm and tmate seems good so we can all get some experience in using upterm and can still fall back to tmate if upterm has issues.

Doing so will also keep the code changes small at this point in time, while we are still evaluating upterm as a replacement.

In the future, we can remove tmate from the file and rename `tmate_debug` to `upterm_debug` -- assuming upterm is our long term solution

Reply:
I am suggesting keep the file and directory names the same for now so we can keep this changeset small. We haven't solidified on upterm yet - though it seems promising.

Reply:
@BlaineEXE , I changed the patch now to add upterm debugging to the existing `tmate_debug/action.yml` instead of adding a new `upterm_debug/action.yml` as you requested.

But even after pondering for a while I did  not find a sensible  way to integrate the existing `secrets.USE_TMATE` mechanism  with the new upterm flavor, so I ended up removing it.
Essentially, `use-tmate` and `use-upterm` are now hard-coded in the `canary-integration-test` workflow when invoking the tmate_debug action, to select the desired flavor.


This way, it should be much easier in the future to restrict debugging to one mechanism when the time has come.